src/smart-components/role/RolesTable.tsx

```diff
@@ -0,0 +1,140 @@
+import { DataViewTable } from '@patternfly/react-data-view/dist/dynamic/DataViewTable';
+import { PageSection, Pagination } from '@patternfly/react-core';
+import { ActionsColumn } from '@patternfly/react-table';
+import ContentHeader from '@patternfly/react-component-groups/dist/esm/ContentHeader';
+import { fetchRoles } from '../../redux/actions/role-actions';
+import { useIntl } from 'react-intl';
+import messages from '../../Messages';
+import { mappedProps } from '../../helpers/shared/helpers';
+import { Role } from '../../redux/reducers/role-reducer';
+import { RBACStore } from '../../redux/store';
+
+const intl = useIntl();
+
+const COLUMNS = [
+  intl.formatMessage(messages.name),
+  intl.formatMessage(messages.description),
+  intl.formatMessage(messages.permissions),
+  intl.formatMessage(messages.workspaces),
+  intl.formatMessage(messages.userGroups),
+  intl.formatMessage(messages.lastModified),
+];
+
+const ROW_ACTIONS = [
+  { title: 'Edit role', onClick: () => console.log('Editing role') },
+  { title: 'Delete role', onClick: () => console.log('Removing role') },
+];
+
+const PER_PAGE = [
+  { title: '5', value: 5 },
+  { title: '10', value: 10 },
+  { title: '20', value: 20 },
+  { title: '50', value: 50 },
+  { title: '100', value: 100 },
+];
+
+// interface Role {
+//   name: string;
+//   description: string;
+//   permissions: string;
+//   workspaces: string[];
+//   userGroup: string;
+//   lastModified: string;
+// }
+
+// const mockRoles: Role[] = [
+//   { name: 'one', description: 'yup', permissions: 'yah', workspaces: ['ok', 'great'], userGroup: 'group1', lastModified: 'today' },
+//   { name: 'two', description: 'yup', permissions: 'yah', workspaces: ['ok', 'great'], userGroup: 'group2', lastModified: 'today' },
+//   { name: 'three', description: 'yup', permissions: 'yah', workspaces: ['ok', 'great'], userGroup: 'group3', lastModified: 'today' },
+//   { name: 'four', description: 'yup', permissions: 'yah', workspaces: ['ok', 'great'], userGroup: 'group4', lastModified: 'today' },
+// ];
+
+const ouiaId = 'RolesTable';
+
+const RolesTable: React.FunctionComponent = () => {
+  // const [roles, setRoles] = useState(mockRoles);
+  const { isLoading, roles } = useSelector((state: RBACStore) => state.roleReducer);
+  // const [totalCount, setTotalCount] = useState(0);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(null);
+
+  const dispatch = useDispatch();
+
+  const pagination = useDataViewPagination({ perPage: 20 });
+  const { page, perPage } = pagination;
+
+  const selection = useDataViewSelection({ matchOption: (a, b) => a === b });
+  const { selected, onSelect, isSelected } = selection;
+
+  const fetchData = useCallback(
+    (apiProps: { limit: number; offset: number; orderBy: string }) => {
+      const { limit, offset, orderBy } = apiProps;
+      dispatch(fetchRoles({ ...mappedProps({ limit, offset, orderBy }), usesMetaInURL: true }));
```

Comment:
I think we should be able to go without `usesMetaInURL` with the new URL management in DataView. It was needed when we managed that in RBAC before